Dockerfiles/agent/amd64/Dockerfile

```diff
@@ -37,8 +38,10 @@ WORKDIR /output
 # Hack: PGP key store locally as it's currently not available from https://keybase.io/justcontainers/key.asc
 COPY hack/s6-pgp-key /tmp/s6-pgp-key
 ENV S6_VERSION v1.22.1.0
-ADD https://github.com/just-containers/s6-overlay/releases/download/${S6_VERSION}/s6-overlay-amd64.tar.gz /output/s6.tgz
-ADD https://github.com/just-containers/s6-overlay/releases/download/${S6_VERSION}/s6-overlay-amd64.tar.gz.sig /tmp/s6.tgz.sig
+ENV JUST_CONTAINERS_DOWNLOAD_LOCATION=${GENERAL_ARTIFACTS_CACHE_BUCKET_URL:+${GENERAL_ARTIFACTS_CACHE_BUCKET_URL}/s6-overlay}
+ENV JUST_CONTAINERS_DOWNLOAD_LOCATION=${JUST_CONTAINERS_DOWNLOAD_LOCATION:-https://github.com/just-containers/s6-overlay/releases/download}
+ADD ${JUST_CONTAINERS_DOWNLOAD_LOCATION}/${S6_VERSION}/s6-overlay-amd64.tar.gz /output/s6.tgz
+ADD ${JUST_CONTAINERS_DOWNLOAD_LOCATION}/${S6_VERSION}/s6-overlay-amd64.tar.gz.sig /tmp/s6.tgz.sig
```

Comment:
We could add checksum checks here.

Reply:
So I guess we can do one of two things here:

* Keep checking the `.sig` signature
* Start checking the tarball checksum (at which point we could actually drop checking the `.sig` signature, because it's redundant).

Thinking about this further, I think I agree with you. If someone successfully attacked the cache, they could replace both the tarball and it's signature file, so the signature check doesn't do much for us here. However, if we hardcode the checksum in the Dockerfile, an attacker would have to both compromise the cache *and* rewrite the checksum in the Dockerfile, which is much more unlikely. So 👍 - I'll drop the `.sig` file and just do checksum validation.